### ISSUE TITLE
[Feature request] SONOFF iHost - SonoffLAN Add-on?

### ISSUE THREAD

**Issue by Hedda** [AlexxIT/SonoffLAN#1144]
Suggest that you consider maybe looking into perhaps creating a SonoffLAN Add-on for the upcoming SONOFF iHost from ITead.

* https://itead.cc/product/sonoff-ihost-smart-home-hub
  * https://sonoff.tech/product-review/sonoff-ihost-smart-home-hub
* https://ewelinkcommunity.net/device-lists/local-server/sonoff-ihost

Note that this suggested feature request is a little early since the product is still only available as pre-order or invited reviewers and devs.

The device is being sold as a local smart home hub and home server as is currently primarily being marketed as a local access and local control device that does not need cloud connection, though I understand that ironically you currently need to sign up for an eWeLink cloud account and login with it in order to enable developer mode on the device using the officially supported procedure for it.

They will initially sell two models referred to as “4G version” (Quad-Core CPU + 4GB RAM) and “2G version” (Dual-Core CPU + 2GB RAM).

It features the same Silicon Labs EFR32MG21 chip as the original Sonoff ZBBridge (and their new Sonoff ZBDongle-E USB dongle):

* https://www.cnx-software.com/2023/03/09/sonoff-ihost-smart-home-hub-enables-local-control-of-sonoff-tasmota-matter-home-automation-devices/

For reference, ITead/Sonoff have official documentation here for their officially supported OS and firmware that it ships with by default as their own operating system does offer official support for locally running third-party applications as "add-ons" via Docker Containers (Micro SD-card may be needed as the device does not have a lot of onboard flash storage), as well as external local access third-party integrations via an official open API (eWeLink CUBE REST API local interface via HTTP requests) that is provided by Sonoff/ITead:

* https://sonoff.tech/ihost-user-guides/
  * https://sonoff.tech/ihost-user-guides/#Local-Control
    * https://sonoff.tech/ihost-user-guides/#Login-and-OTA
  * https://sonoff.tech/ihost-user-guides/add-on/#Add-on
  * https://sonoff.tech/ihost-user-guides/api/
    * https://ewelink.cc/ewelink-cube/
      * https://ewelink.cc/ewelink-cube/open-api/

More references:
* https://itead.cc/wp-content/uploads/2023/03/Core-Board-Finger-M.2-M-267pin-External-Interface-Definition.pdf
* https://notenoughtech.com/home-automation/sonoff-ihost/
  * https://www.youtube.com/watch?v=KBvGWoWq__Y
* https://ewelinkcommunity.net/2023/03/sonoff-places-teaser-video-of-upcoming-ihost-online-what-can-we-expect-from-this-device/
* https://ewelinkcommunity.net/2023/03/sonoff-released-their-ihost-affordable-computer-to-completely-make-your-smarthome-locally-controlled/

PS: FYI, additional technical details which could be used to hack it further than officially supported by Sonoff/ITead being collected here:

https://community.home-assistant.io/t/sonoff-ihost-smart-home-hub-computer-from-itead-can-it-be-hacked-to-run-home-assistant-os-with-bluetooth-thread-matter-and-zha-integrations/545962

**Comment by eerke2021** [comment]
Is English your first language?

**Comment by Hedda** [comment]
> Is English your first language?

No.

**Comment by eerke2021** [comment]
Thought so. On 31 Mar 2023, at 19:59, Hedda ***@***.***> wrote:﻿

Is English your first language?

No.

**Comment by zvrka** [comment]
would realy like to se add on for ihost made by Alexx as its integration is much better then official one for HA.

**Comment by eerke2021** [comment]
Even if it could be installed in a docker on the SD card mine currently randomly decides whether there is an SD card now anyway. Tried two cards, both of which work in my PC. Their website says that USB storage is possible, but I’ve tested it and it just hangs if you put a memory stick there. Nothing happened when I put an external SD card reader in the slot. To work as a security device it really needs reliable storage. On 13 Apr 2023, at 01:08, zvrka ***@***.***> wrote:﻿
would realy like to se add on for ihost made by Alexx as its integration is much better then official one for HA.

**Comment by zvrka** [comment]
> Even if it could be installed in a docker on the SD card mine currently randomly decides whether there is an SD card now anyway. Tried two cards, both of which work in my PC. Their website says that USB storage is possible, but I’ve tested it and it just hangs if you put a memory stick there. Nothing happened when I put an external SD card reader in the slot. To work as a security device it really needs reliable storage. On 13 Apr 2023, at 01:08, zvrka ***@***.***> wrote:﻿ would realy like to se add on for ihost made by Alexx as its integration is much better then official one for HA. —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

mine card works perfectly,you did not read that right...usb port on ihost for now is only for debuging purposses

**Comment by eerke2021** [comment]
Nope. This page on the ITEAD says these exact words:SONOFF iHost Smart Home Hub - SONOFF Officialsonoff.tech“Storage Methods: USB and SD card (max. 256 GB)“It would be pretty hard to misread that. On 13 Apr 2023, at 01:44, zvrka ***@***.***> wrote:﻿

Even if it could be installed in a docker on the SD card mine currently randomly decides whether there is an SD card now anyway. Tried two cards, both of which work in my PC. Their website says that USB storage is possible, but I’ve tested it and it just hangs if you put a memory stick there. Nothing happened when I put an external SD card reader in the slot. To work as a security device it really needs reliable storage. On 13 Apr 2023, at 01:08, zvrka @.> wrote:﻿ would realy like to se add on for ihost made by Alexx as its integration is much better then official one for HA. —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: @.>

mine card works perfectly,you did not read that right...usb port on ihost for now is only for debuging purposses

**Comment by AlexxIT** [comment]
The Coolkit team was kind enough to send me this device (all other Sonoff devices I've always bought myself).
I'll see what I can do when I get it.

**Comment by eerke2021** [comment]
Mine keeps forgetting it has an SD card, but assuming yours works there's expandable storage, it's Linux based and can run dockers. I believe there is also an open local API, but from memory that's being released in June on their schedule, although I imagine they'd let you know before then. I've not read anywhere what information is communicated to docker containers from the main body or what antennas are available directly, although there are already instructions on installing node-red and homebridge on their website.

**Comment by sipimokus** [comment]
Hello, 

I have started developing a pip package for the eWelink CUBE API, which is used by iHost (and NSPanel Pro as well). 
It's created about from the existing Node-RED package, code is available at here:
https://github.com/sm4rth0m3/python-pip.sonoff-ewelink-cube-client-api

This is just a REST API library and does not include integration with HASS, but I can try keep into consideration (using asyncio and aiohttp).

I hope it will be useful.

**Comment by unnamedi** [comment]
Is there any update on this?
It would be magical to be able to see what's on the iHost without using HomeKit
